src/EditorFeatures/VisualBasic/UseAutoProperty/UseAutoPropertyAnalyzer.vb

```diff
@@ -72,12 +72,11 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UseAutoProperty
             ' code had different accessibility between the accessors and property then we 
             ' can't convert it.
 
-            ' getter has to match property accessibility
-            If prop.GetMethod.DeclaredAccessibility <> prop.DeclaredAccessibility Then
+            If prop.GetMethod IsNot Nothing AndAlso
```

Comment:
is this covered by tests?

Reply:
@CyrusNajmabadi when you added this line was that because of a failing test? or would this be an untested codepath?

Reply:
I added it just to be conservative.  Technically there should always be a get method due to previous checks in the base class.  But i still want to just have this code be simple and not dependent on ordering of the caller.

Reply:
I don't like that we're adding untested code. If it's possible for this code to hit, add a test for it. If it's the case that this can't happen due to previous checks, add an assert. That way you'll also make it clear what the assumptions are here. Same for the setter below.

If you leave it like this, someone is going to come along a few months later, notice that it's unnecessary and delete it anyway. Or they'll end up in the same situation that I was here: https://github.com/dotnet/roslyn/pull/25394/files#r173603729 where the solution is "well, this shouldn't happen, but let's not touch the code and just assume that whoever wrote this knew that for some magical reason it can happen but didn't bother writing tests" (and that doesn't sound like a long-term solution - in fact that sounds like the definition of legacy code to me, and you're creating the same situation here).

At the very least if you do something like this - adding a piece of code without a test and without an obvious reason, add a comment explaining either why it's necessary even though all the tests pass even if you remove this (and maybe also apologize for not writing the test) or saying that it's not necessary but you added it "just to be sure" (which to me doesn't really sound like a legitimate reason, but at least I know that happened here).

Reply:
> But i still want to just have this code be simple and not dependent on ordering of the caller

Even if you consider this to be a helper that shouldn't depend on the rest of the code, it is still wrong. If you have no getter and no setter, this will return true, which is obviously wrong for 'CanConvert'. This isn't really a general purspose helper and can't be thought of as one. It depends on the checks done elsewhere in this class/the base class either way.

Reply:
Sorry for sounding so opinionated, I'm making too much of a big thing out of this. It's ok if you leave it as is but just consider that from my point of view, and maybe other future contributors as well: I can definitely see myself looking at this code and wondering endlessly why the check would be necessary. And then being afraid to touch or change the code at all thinking there must be some hidden reasons for all this and many untested cases that I would unknowingly break.

Reply:
I hear your feedback.  But i'm really ok with it as is.  

> I can definitely see myself looking at this code and wondering endlessly why the check would be necessary.

I think this code is sufficient.  It explains what's going on (namely that VB cannot create an auto-prop where the get or set differs from the property), and then it tests for just that.  Seems simple, reasonable, and not something that would truly be that confusing to figure at at some future point.